interface/src/app/tribe_client/genesets.js

```diff
@@ -0,0 +1,88 @@
+angular.module('adage.tribe_client.genesets', [
+  'adage.tribe_client.resource'
+])
+
+// Geneset Search Factory
+// Allows for retreiving for and interacting with search results for genesets.
+.factory('GenesetSearch', function($rootScope, Genesets) {
+  var genesets = [];
+  var query = {};
+  var totalResults = 0;
+
+  return {
+    getQuery: function() {
+      return query;
+    },
+    getGenesets: function() {
+      return genesets;
+    },
+    clear: function() {
+      query = {};
+      genesets = [];
+      $rootScope.$broadcast('genesets.update');
+    },
+    totalResults: function() {
+      return totalResults;
+    },
+    query: function(searchParams) {
+      query = searchParams;
+      Genesets.query(query, function(data) {
+        totalResults = data.meta.total_count;
+        genesets = data.objects;
+        $rootScope.$broadcast('genesets.update');
```

Comment:
Once `genesets` is bound to the parent's scope via the `.directive()` modifications I suggested, this `$broadcast` can be eliminated.